homeassistant/components/withings/common.py

```diff
@@ -751,19 +752,19 @@ async def async_get_all_data(self) -> dict[MeasureType, Any] | None:
                         for flow in self._hass.config_entries.flow.async_progress_by_handler(
                             const.DOMAIN
                         )
-                        if flow.context == context
+                        if flow["context"].get(const.PROFILE) == self._profile
                     ),
                     None,
                 )
                 if flow:
-                    return
+                    return None
 
                 # Start a reauth flow.
                 await self._hass.config_entries.flow.async_init(
```

Comment:
It would be better to replace all the logic here by just calling `config_entry.async_start_reauth`.

https://github.com/home-assistant/core/blob/0d6bf08ff69a0b75ffe797f49c201d1b7c4f3a72/homeassistant/config_entries.py#L641-L661

It should be enough to match on config entry entry_id and unique_id and add the profile from the config entry data to the flow context in the reauth step.

Reply:
The trouble with using `async_start_reauth` was that the current implementation passes the profile in the `context`, and the whole config_flow assumes the profile will be passed that way.

Now the change is much bigger... and maybe riskier